src/Sylius/Bundle/CoreBundle/EventListener/TaxonDeletionListener.php

```diff
@@ -73,4 +73,15 @@ public function removeTaxonFromPromotionRules(GenericEvent $event): void
             ]);
         }
     }
+
+    public function preventRemovingRootTaxonAtPositionZero(GenericEvent $event): void
```

Comment:
This name is a bit confusing, it might suggest that you cannot remove the taxon at all.
I'd be for either `handleRemovingRootTaxonAtPositionZero` or something like that.